### PR TITLE
segmentation_models.pytorch integration

### LINKED ISSUE #515: Modify SEMANTIC_SEGMENTATION_HEADS and SEMANTIC_SEGMENTATION_BACKBONES registries to work with segmentation_models.pytorch

### PR DESCRIPTION
**Pull request by aniketmaurya**
## What does this PR do?

Fixes #515 #516

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/image/segmentation/backbones.py [hunk 1]
```diff
@@ -14,12 +14,17 @@
...
 from flash.core.registry import FlashRegistry
-from flash.core.utilities.imports import _TORCHVISION_AVAILABLE
+from flash.core.utilities.imports import _SEGMENTATION_MODELS_AVAILABLE, _TORCHVISION_AVAILABLE
 from flash.image.backbones import catch_url_error
 
 if _TORCHVISION_AVAILABLE:
     from torchvision.models import mobilenetv3, resnet
 
+if _SEGMENTATION_MODELS_AVAILABLE:
+    import segmentation_models_pytorch as smp
```
Comment (tchaton) [inline flash/image/segmentation/backbones.py:24]: Let's move `ENCODERS` to the bottom, so model imports and registry population are done together.

File: flash/image/segmentation/heads.py [hunk 2]
```diff
@@ -11,33 +11,65 @@
...
+            arch=head,
+            encoder_name=backbone,
+            encoder_weights=encoder_weights,
+            classes=num_classes,
+            in_channels=in_channels,
+            **kwargs,
+        )
+
+    for model_name in SMP_MODELS:
+        SEMANTIC_SEGMENTATION_HEADS(
```
Comment (tchaton) [inline flash/image/segmentation/heads.py:66]: Neat !

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by ethanwharris — APPROVED** [review]
Awesome! LGTM :smiley:

**Pull request merged**